backend/benefit/common/utils.py

```diff
@@ -31,8 +35,82 @@ def xgroup(iter, n=2, check_length=False):
             last = []
 
 
+def pairwise(iterable):
+    "s -> (s0,s1), (s1,s2), (s2, s3), ..."
+    a, b = itertools.tee(iterable)
+    for elem in b:
```

Comment:
What is this for loop doing?

Reply:
it removes the first element from b, so that in zip(a,b) the iterable a always returns the preceding element